src/Data/Nat/Divisibility/Core.agda

```diff
@@ -35,7 +35,37 @@ record _∣_ (m n : ℕ) : Set where
   constructor divides
   field quotient : ℕ
         equality : n ≡ quotient * m
-open _∣_ using (quotient) public
+
+  quotient≡0⇒n≡0 : quotient ≡ 0 → n ≡ 0
+  quotient≡0⇒n≡0 q≡0 = subst (λ q → n ≡ q * m) q≡0 equality
+
+  instance
+    quotient≢0 : .{{NonZero n}} → NonZero quotient
```

Comment:
Doesn't this instance run into the problems discussed in https://github.com/agda/agda-stdlib/issues/1565

Reply:
Hmmm... not sure!?
I confess I haven't really understood much of that discussion or the examples.
In the one intended candidate specimen deployment I've looked at (your `factorise` function), the instance is only brought into scope by an explicit `open`... so perhaps the thing to do is restrict the export from the module?

Reply:
Latest commit takes the second approach...